scripts/ubuntu/1-install-dependendcies.sh

```diff
@@ -17,10 +17,42 @@
 set -e
 
 # Install Docker Package Repo
-curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor --yes -o /usr/share/keyrings/docker-archive-keyring.gpg
+echo
+echo "####################################"
+echo "Install Docker Package Repo"
+echo "####################################"
+echo
+timeout 300s bash -c '
+start_time=$(date)
+echo "Ping started at: $start_time"
+while :; do
+  if ping -c 1 docker.download.com | grep -q "1 received"; then
+    echo "Ping docker.download.com successful"
+    end_time=$(date)
+    echo "Ping ended at: $end_time"
+    echo "Ping duration: $(($(date +%s) - $(date -d "$start_time" +%s))) seconds"
+    break
+  fi
+  echo "Ping docker.download.com failed, retrying..."
+  sleep 1
+done
+if [ $? -eq 124 ]; then
+  end_time=$(date)
+  echo "docker.download.com: Timeout reached"
+  echo "Ping ended at: $end_time"
+  echo "Ping duration: $(($(date +%s) - $(date -d "$start_time" +%s))) seconds"
+fi
+'
+
+curl -fsSL --retry 2 --retry-connrefused -v https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor --yes -o /usr/share/keyrings/docker-archive-keyring.gpg
```

Comment:
Maybe the PR description is outdated, but then we don't need to use the `-4fsSL` anymore?
The difference here is the retry options plus the ping test before?

Reply:
This line has been restored to original